client/state/posts/selectors.js

```diff
@@ -20,10 +32,40 @@ import { DEFAULT_POST_QUERY } from './constants';
  * @return {Object}          Post object
  */
 export function getPost( state, globalId ) {
-	return state.posts.items[ globalId ];
+	return getPostsByGlobalId( state )[ globalId ];
+}
+
+/**
+ * Returns an array of all posts.
+ *
+ * @param  {Object} state    Global state tree
+ * @return {Array}           Posts array
+ */
+export function getPosts( state ) {
+	return state.posts.items;
 }
 
 /**
+ * Returns a mapping of site ID, post ID pairing to global post ID.
+ *
+ * @param state
+ */
+export const getSitePosts = createSelector(
```

Comment:
I didnt go deep into the reSelect library, but do we have a way to move `createSelector` call to the vicinity of `connect` ?
What I mean is, I liked the pure syntax of selectors and createSelector seems like some magical wrapper, just like connect.
Does it make sense or not so much?